src/strategies/GammaUniswapV3MerklFarmStrategy.sol

```diff
@@ -303,35 +304,81 @@ contract GammaUniswapV3MerklFarmStrategy is LPStrategyBase, MerklStrategyBase, F
+        IFactory.Farm memory farm = _getFarm();
+
+        if (farm.nums[0] == ALMPositionNameLib.STABLE) {
+            handleStableAmounts(amountsMax, $, underlying_, _assets, amount1Start, amount1End, amountsConsumed);
+        } else {
+            handleNonStableAmounts(amountsMax, $, underlying_, _assets, amount1Start, amount1End, amountsConsumed);
+        }
+
+        // calculate shares
+        value = calculateShares(amountsConsumed, underlying_);
+    }
+
+    function handleStableAmounts(
+        uint[] memory amountsMax,
+        GammaUniswapV3FarmStrategyStorage storage $,
+        address underlying_,
+        address[] memory _assets,
+        uint amount1Start,
+        uint amount1End,
+        uint[] memory amountsConsumed
+    ) internal view {
+        amountsConsumed[1] = amountsMax[1];
+        amountsConsumed[0] = amountsMax[0];
+        (, uint amount0End) = $.uniProxy.getDepositAmount(underlying_, _assets[1], amountsMax[1]);
+
+        // Inline the assignment and condition with a ternary operator
+        amountsConsumed[1] = (amountsMax[1] > amount1End) ? amount1End : amountsMax[1];
+
+        // Check the second condition within another ternary operation
+        (amountsMax[1] <= amount1Start) ? amountsConsumed[0] = amount0End : amountsConsumed[0];
+
+        // Set amountsConsumed[1] to amountsMax[1] only if the second condition holds true
+        amountsConsumed[1] = (amountsMax[1] <= amount1Start) ? amountsMax[1] : amountsConsumed[1];
+
+        // Ensure amountsConsumed[0] does not exceed amount0End
+        amountsConsumed[0] = (amountsConsumed[0] > amount0End) ? amount0End : amountsConsumed[0];
+    }
+
+    function handleNonStableAmounts(
+        uint[] memory amountsMax,
+        GammaUniswapV3FarmStrategyStorage storage $,
+        address underlying_,
+        address[] memory _assets,
+        uint amount1Start,
+        uint amount1End,
+        uint[] memory amountsConsumed
+    ) internal view {
         if (amountsMax[1] > amount1End) {
             amountsConsumed[0] = amountsMax[0];
-            // its possible to be (amount1End + amount1Start) / 2, but current amount1End value pass tests with small amounts
             amountsConsumed[1] = amount1End;
         } else if (amountsMax[1] <= amount1Start) {
-            //slither-disable-next-line similar-names
             (uint amount0Start, uint amount0End) = $.uniProxy.getDepositAmount(underlying_, _assets[1], amountsMax[1]);
             amountsConsumed[0] = (amount0End + amount0Start) / 2;
             amountsConsumed[1] = amountsMax[1];
         } else {
             amountsConsumed[0] = amountsMax[0];
             amountsConsumed[1] = amountsMax[1];
         }
+    }
 
-        // calculate shares
+    function calculateShares(uint[] memory amountsConsumed, address underlying_) internal view returns (uint value) {
         IHypervisor hypervisor = IHypervisor(underlying_);
-        //slither-disable-next-line unused-return
```

Comment:
dont remove slither and semgrep alerts mitigation

Reply:
Ok. added that.